examples/wifi/nodemcu/waziup/code

```diff
@@ -0,0 +1,117 @@
+#include <SPI.h>
+#include <ESP8266WiFi.h>
+#include <ArduinoHttpClient.h>
+
+char ssid[] = "*******";   // your network SSID (name)
+char pass[] = "***********";    // your network password (use for WPA, or use as key for WEP)
+
+// Déclaration path and port
+char server[] = "api.waziup.io";
+char resource[]   = "/api/v2/devices/Water/ultrasonic/Distance/distance";
+char contentType[] = "text/plain;charset=utf-8";
+int port = 80;
+
+WiFiClient wifiClient;
+HttpClient httpClient = HttpClient(wifiClient, server, port);
+
+// defines pins numbers
+const int trigPin = 2;  //D4
+const int echoPin = 0;  //D3
+
+
+
+void setup() {
+  
+  pinMode(trigPin, OUTPUT); // Sets the trigPin as an Output
+  pinMode(echoPin, INPUT); // Sets the echoPin as an Input
+
+  Serial.begin(115200); //baud rate ,if you want to see the process in the serial monitor ,same baud rate should be set.
+  //Initialize serial and wait for port to open:
+//  Serial.begin(9600);
+  while (!Serial) {
+    ; // wait for serial port to connect. Needed for native USB port only
+  }
+  Serial.println("Ready");
+  // check for the WiFi module:
+  if (WiFi.status() == WL_NO_SHIELD) {
+    Serial.println("Communication with WiFi module failed!");
+    // don't continue
+    while (true);
+  }
+
+  // attempt to connect to WiFi network:
+  int status = WL_IDLE_STATUS;
+  while (status != WL_CONNECTED) {
+    Serial.print("Attempting to connect to SSID: ");
+    Serial.println(ssid);
+    // Connect to WPA/WPA2 network. Change this line if using open or WEP network:
+    status = WiFi.begin(ssid, pass);
+    // wait 10 seconds for connection:
+    delay(10000);
+  }
+
+  Serial.println("Connected to wifi");
+  printWiFiStatus();
+}
+
+void loop() {
+  // defines variables
+  float duration;
+  float distance;
+  // Clears the trigPin
+  digitalWrite(trigPin, LOW);
+  delayMicroseconds(2);
+
+  // Sets the trigPin on HIGH state for 10 micro seconds
+  digitalWrite(trigPin, HIGH);
+  delayMicroseconds(10);
+  digitalWrite(trigPin, LOW);
+
+  // Reads the echoPin, returns the sound wave travel time in microseconds
+  duration = pulseIn(echoPin, HIGH);
+
+  // Calculating the distance
+  distance = duration * 0.034 / 2;
+
+
+
+
+
+  //Send the message to WaziCloud
+  httpClient.post(resource, contentType, "distance");
```

Comment:
You should remove the quote to "distance", to get the variable. Otherwise, it's just a String.